### PR TITLE
:book: Document Versioning in contributing guide

### PR DESCRIPTION
**Pull request by kashifest**
This changeset also completely removes the outdated VERSIONING.md document. We follow the CAPI workflow in this regard which is documented here: https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,6 +33,47 @@ sounds interesting](https://github.com/metal3-io/cluster-api-provider-metal3/iss
...
+
+#### Backporting
+
+We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+backport MUST not be breaking for either API or behavioral changes.
+We generally do not accept PRs against older release branches.
+
+## Branches
+
+Cluster API Metal3 has two types of branches: the *main* branch and
```
Comment (furkatgofurov7) [inline CONTRIBUTING.md:65]:
```suggestion
Cluster API Provider Metal3 has two types of branches: the *main* branch and
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -57,7 +104,7 @@ mistakes](https://github.com/golang/go/wiki/CodeReviewComments) in your PRs.
 
 ## Backporting a Patch
 
-Cluster API maintains older versions through `release-X.Y` branches. We accept
+Cluster API Metal3 maintains older versions through `release-X.Y` branches. We accept
```
Comment (furkatgofurov7) [inline CONTRIBUTING.md:107]:
```suggestion
Cluster API Provider Metal3 maintains older versions through `release-X.Y` branches. We accept
```

**Comment by furkatgofurov7** [comment]
Also, please fix the markdown issues reported in the CI

**Comment by kashifest** [comment]
> Also, please fix the markdown issues reported in the CI

Thanks for the review. PTAL again.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -33,6 +33,47 @@ sounds interesting](https://github.com/metal3-io/cluster-api-provider-metal3/iss
 Alternatively, read some of the docs on other controllers and try to write your
 own, file and fix any/all issues that come up, including gaps in documentation!
 
+## Versioning
+
+### Codebase and Go Modules
+
+> :warning: The project does not follow Go Modules guidelines for compatibility requirements for 1.x semver releases.
+Cluster API Metal3 follows Cluster API release cadence abd versioning which follows upstream Kubernetes semantic versioning. With the v1 release of our codebase, we guarantee the following:
```
Comment (furkatgofurov7) [inline CONTRIBUTING.md:41]:
```suggestion
Cluster API Provider Metal3 follows Cluster API release cadence and versioning which follows upstream Kubernetes semantic versioning. With the v1 release of our codebase, we guarantee the following:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -33,6 +33,47 @@ sounds interesting](https://github.com/metal3-io/cluster-api-provider-metal3/iss
...
+These guarantees extend to all code exposed in our Go Module, including
+*types from dependencies in public APIs*.
+Types and functions not in public APIs are not considered part of the guarantee.
+The test module, and experiments do not provide any backward compatible guarantees.
+
+#### Backporting
+
+We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+backport MUST not be breaking for either API or behavioral changes.
+We generally do not accept PRs against older release branches.
```
Comment (furkatgofurov7) [inline CONTRIBUTING.md:61]: Q: I remember this being discussed recently upstream, however not sure if we should really follow it as a provider? https://github.com/kubernetes-sigs/cluster-api/pull/7167

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks, last nit that we could fix:

File: CONTRIBUTING.md (hunk 3 above)
Comment (kashifest) [reply CONTRIBUTING.md]: Thanks fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (kashifest) [reply CONTRIBUTING.md:61]: I would prefer to keep it as it is now.

**Review by furkatgofurov7 — APPROVED** [review]

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**